src/Suggestor/DependencyHelpers.ts

```diff
@@ -54,9 +54,9 @@ function searchDescriptionWithoutTags(query: string, allTasks: Task[]): Task[] {
 export function searchForCandidateTasksForDependency(
     search: string,
     allTasks: Task[],
-    task: Task,
-    blockedBy: Task[],
-    blocking: Task[],
+    task?: Task,
+    blockedBy?: Task[],
+    blocking?: Task[],
```

Comment:
Why are these now optional?

Reply:
See Comment: https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2771#discussion_r1564651869

Superficial Reason being, that those can be "undefined"/"null".
(For Example for Testing, Most Easy Quick Fix for now)

The Root Problem of it lies in the Structure of the Task Class, and the Caching Behavior, in my Opinion.
This should be refactored. Unfortunate this is out of the Scope for me, for now, to put more Thinking & Time in.
That's why i choose to do it that way.